core/api/src/main/java/org/trellisldp/api/ResourceService.java

```diff
@@ -117,6 +118,18 @@ default RDFTerm unskolemize(final RDFTerm term) {
         return term;
     }
 
+    /**
+     * Update the modification date of the provided resource.
+     *
+     * @apiNote there may be cases where the resource with the given identifier does not (yet) exist.
+     *          In those cases, the implementation should return a completed future as if the resource
```

Comment:
Wait, whoa-- if there is no resource, this method returns as though there was? I guess that's not incorrect, but it seems odd... is there an impl reason for this?

I hate to say it, but if this returned `CompletableFuture<? extends Resource>` we could return `MISSING_RESOURCE` in that case.

Reply:
Actually, I wrote that note before wiring it up in the HTTP layer (and before writing the triplestore implementation), and in practice, there will always be an existing resource. There _may_ be some corner cases, such as a child resource is added to a direct container at the same time that the associated membership resource gets deleted, but that may be outside the scope of what Trellis manages.

Basically, the way this is wired in the HTTP layer follows the same logic that generates notifications, which involves some level of verification that the parent/membership resources exist in the first place.

Which is to say, I should really revise or remove that note. What I was trying to express is that, given that Trellis doesn't enforce a transactional model, it is theoretically possible that the resource wouldn't exist. However, if an implementation wanted to throw an exception in the case of a non-existing resource, that wouldn't necessarily be wrong.

Reply:
Okay, this:

>  if an implementation wanted to throw an exception in the case of a non-existing resource, that wouldn't necessarily be wrong.

makes me feel better, and if it's okay, that's what I intend to do in `trellis-cassandra`.

Reply:
I've updated the javadoc for this, which should clarify things.